Remotion/ObjectBinding/Web.Development.WebTesting/ControlObjects/BocControlObject.cs

```diff
@@ -69,39 +69,40 @@ protected IReadOnlyList<string> GetValidationErrors ([NotNull] ElementScope scop
       if (IsValid (scope))
         return new List<string>();
 
-      ElementScope validationErrorsScope;
       string describedBy;
 
       try
       {
-        describedBy = scope["aria-describedby"];
+        describedBy = scope.GetAttributeOrThrow ("aria-describedby");
       }
       catch (MissingHtmlException exception)
       {
-        throw new MissingHtmlException (
-            "Could not find validation error field. This could be due to wrong markup or a missing validators.",
-            exception);
+        throw GetMissingValidationErrorFieldException (exception);
       }
 
       var validationErrorIDs = describedBy.Split (new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
 
+      var validationErrorIndex = int.Parse (scope.GetAttributeOrThrow (DiagnosticMetadataAttributes.ValidationErrorIDIndex));
+      var validationErrorsScope = Scope.FindId (validationErrorIDs[validationErrorIndex]);
+
+      // re-motion renders the delimiter as <br />, but the browser (and Selenium) show it as <br>
+      const string validationErrorDelimiter = "<br>";
+
       try
       {
-        var validationErrorIndex = int.Parse (scope[DiagnosticMetadataAttributes.ValidationErrorIDIndex]);
-
-        validationErrorsScope = Scope.FindId (validationErrorIDs[validationErrorIndex]);
+        return validationErrorsScope.InnerHTML.Split (new[] { validationErrorDelimiter }, StringSplitOptions.RemoveEmptyEntries);
       }
       catch (MissingHtmlException exception)
       {
-        throw new MissingHtmlException (
-            "Could not find validation error field. This could be due to wrong markup or a missing validators.",
-            exception);
+        throw GetMissingValidationErrorFieldException (exception);
       }
+    }
 
-      // re-motion renders the delimiter as <br />, but the browser (and Selenium) show it as <br>
-      const string validationErrorDelimiter = "<br>";
-
-      return validationErrorsScope.InnerHTML.Split (new[] { validationErrorDelimiter }, StringSplitOptions.RemoveEmptyEntries);
+    private MissingHtmlException GetMissingValidationErrorFieldException (MissingHtmlException innerException)

```

Comment:
"Create" not "Get". Factory methods are usually named "Create". Get is only used when it is some kind of lazy-initialized field. The infamous GetOrCreate can be used when the optional value creation should be made obvious to the caller. Usually used with caches or data stores.

Reply:
Fixed in https://github.com/re-motion/Trunk-SVN2Git/pull/11/commits/2ff1c605d3533f58d855b42dc8b92eb82d2126ca.